### PR TITLE
Issue #13691 and #13692 corrections (exp-body ctor/dtor fails to call base())

### PR DESCRIPTION
**Pull request by lachbaer**
The tests for issue #13691 and #13692 go through now.

### CODE REVIEW

**Comment by gafter** [comment]
I'm sorry, I already have a PR in flight for this. I realize yours came first, but I didn't see it until now.

File: src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.cs [hunk 1]
```diff
@@ -536,7 +543,7 @@ internal static BoundBlock ConstructDestructorBody(MethodSymbol method, BoundBlo
                             baseTypeFinalize)
                         )
                     { WasCompilerGenerated = true },
-                    ((BlockSyntax)syntax).CloseBraceToken.Span);
+                    endBodySpan);
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.cs]: >endBodySpan [](start = 20, length = 11)

I don't think it is correct to produce a sequence point in this case.

**Comment by gafter** [comment]
@dotnet/roslyn-compiler Need a second reviewer for this, please.

**Comment by gafter** [comment]
Other than the sequence point thing, this looks good to me.

File: src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.cs (hunk 1 above)
Comment (lachbaer) [inline src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.cs]: You're (probably) right. Please go with your solution (#13765).

**Comment by lachbaer** [comment]
@gafter No problem that you did it, too. I took it as an exercise for myself to find the bug :-) It looks as if our solutions are pretty the same, except that you added tests and your new binding of the finalizers is much more elegant.

**Comment by gafter** [comment]
Actually, I like the way you did the finalizers better ;)

**Comment by gafter** [comment]
@lachbaer Just to be clear, are you handing this off to me at this point?

**Comment by lachbaer** [comment]
@gafter Whatever you suggest and what is better for the project. I'd still like to maintain it, so I'm going to put the changes in my fork and PR.

What do you like better about the finalizers? The code change in BindMethodBody(...)? Because you weren't lucky with producing a sequence point in ConstructDestructorBody(...) and your fix with `BoundStatement baseFinalizeCall = new BoundExpressionStatement(...` seems to make more sense.

**Comment by gafter** [comment]
OK, then lets move forward with this PR.

What I like about the way you handle destructors was not duplicating the code (the if statement in BindMethodBody).

Can you please rebase your changes to the features/exprbody branch and fix the sequence point issue?

**Comment by lachbaer** [comment]
@gafter I haven't merged in your updated tests yet.

File: src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs [hunk 2]
```diff
@@ -1551,34 +1551,32 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
...
+                    if (method.MethodKind != MethodKind.Destructor)
                     {
-                        return MethodBodySynthesizer.ConstructDestructorBody(method, body);
-                    }
-
-                    foreach (var iterator in binder.MethodSymbolsWithYield)
-                    {
-                        foreach (var parameter in iterator.Parameters)
+                        foreach (var iterator in binder.MethodSymbolsWithYield)
                         {
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs]: I think this loop should be run for all methods, destructor or not (I realize it wasn't before).

File: src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs [hunk 3]
```diff
@@ -346,7 +346,7 @@ static void Main()
             CompileAndVerify(source, expectedOutput: "GetAction 1");
         }
 
-        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/13691")]
+        [Fact(DisplayName = "https://github.com/dotnet/roslyn/issues/13691")]
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs]: >DisplayName [](start = 14, length = 11)

We use `WorkItem` for this purpose.

**Comment by gafter** [comment]
@lachbaer I'm going to cherry-pick your changes into a separate PR. Thanks for your help. I'll let you know where this goes.

**Pull request closed without merging**